[GTK] Crash on inputting booking field on IRCTC website in ScriptMessageClientGtk::didPostMessage

mcatanzaro opened the pull request:
#### 55790984429f1c921d2eff6ade9331938cae0364
<pre>
[GTK] Crash on inputting booking field on IRCTC website in ScriptMessageClientGtk::didPostMessage
<a href="https://bugs.webkit.org/show_bug.cgi?id=280150">https://bugs.webkit.org/show_bug.cgi?id=280150</a>

Reviewed by Adrian Perez de Castro.

Epiphany is failing to unregister a script message handler, causing it
to be emitted after its corresponding WebKitWebView and
WebKitUserContentManager have been destroyed. There are basically three
ways we could handle this: (a) detect it and ignore it, (b) detect it
and emit a warning or critical, and (c) proactively prevent it by
storing a HashMap containing all message handlers and unregistering any
that still exist when the WebKitUserContentManager is destroyed.

Maybe (c) would be best, but I&apos;m lazy, so I chose option (b).

I spent a little while trying to create an API test for this, but it
doesn&apos;t seem simple.

* Source/WebKit/UIProcess/API/glib/WebKitUserContentManager.cpp:

Canonical link: <a href="https://commits.webkit.org/285924@main">https://commits.webkit.org/285924@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/860655300e4dedb637cf11cfdbe7b74dc1126aca

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/69242 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/48642 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/21915 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/73324 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/20400 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/56443 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/20249 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/55082 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/13544 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/72308 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/44385 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/59779 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/35561 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/41055 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/18774 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/62997 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/17554 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/75036 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/13224 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/16782 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/62756 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/13263 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/59862 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/62662 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/16346 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/10657 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/4274 "Passed tests") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/44446 "Built successfully") | [⏳ 🛠 mac-safer-cpp ](https://ews-build.webkit.org/#/builders/macOS-Safer-CPP-Checks-EWS "Waiting in queue, processing has not started yet") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/45520 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/46715 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/45261 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aperezdc reviewed (approved):
Sorry this took me a while to check, it was under a few piles of bugmail. Approved!

mcatanzaro committed:
[GTK] Crash on inputting booking field on IRCTC website in ScriptMessageClientGtk::didPostMessage
https://bugs.webkit.org/show_bug.cgi?id=280150

Reviewed by Adrian Perez de Castro.

Epiphany is failing to unregister a script message handler, causing it
to be emitted after its corresponding WebKitWebView and
WebKitUserContentManager have been destroyed. There are basically three
ways we could handle this: (a) detect it and ignore it, (b) detect it
and emit a warning or critical, and (c) proactively prevent it by
storing a HashMap containing all message handlers and unregistering any
that still exist when the WebKitUserContentManager is destroyed.

Maybe (c) would be best, but I'm lazy, so I chose option (b).

I spent a little while trying to create an API test for this, but it
doesn't seem simple.

* Source/WebKit/UIProcess/API/glib/WebKitUserContentManager.cpp:

Canonical link: https://commits.webkit.org/285924@main
